### PR TITLE
[DEN-379] Add terraform support for CPS Certificates - CREATE certificate, add notification fields

### PR DESCRIPTION
**Pull request by stevenpaz**

### CODE REVIEW

File: edgecast/resources/cps/certificate.go [hunk 1]
```diff
@@ -47,31 +47,61 @@ func ResourceCertificateCreate(
 		return diag.Errorf("failed to load configuration")
 	}
 
-	cpsService, err := buildCPSService(**config)
+	ns, errs := ExpandNotificationSettings(d.Get("notification_setting"))
```
Comment (wdawg33) [inline edgecast/resources/cps/certificate.go:50]: Did you want to check for errs here and return an error?

File: edgecast/resources/cps/certificate.go [hunk 2]
```diff
@@ -47,31 +47,61 @@ func ResourceCertificateCreate(
...
+	if err != nil {
+		d.SetId("")
+		return diag.FromErr(err)
+	}
+
+	//ns, errs := ExpandNotificationSettings(d.Get("notification_settings"))
 
-	resp, err := cpsService.Certificate.CertificatePost(params)
+	if len(errs) > 0 {
+		return helper.DiagsFromErrors(errs)
```
Comment (wdawg33) [inline edgecast/resources/cps/certificate.go:75]: We should SetId("") here. There are also a few other places in this function we should be as well if you want to add them..

File: edgecast/resources/cps/certificate.go [hunk 3]
```diff
@@ -47,31 +47,61 @@ func ResourceCertificateCreate(
...
+	cparams := certificate.NewCertificatePostParams()
+	cparams.Certificate = cert
+
+	cresp, err := svc.Certificate.CertificatePost(cparams)
+	if err != nil {
+		d.SetId("")
+		return diag.FromErr(err)
+	}
+
+	//ns, errs := ExpandNotificationSettings(d.Get("notification_settings"))
```
Comment (wdawg33) [inline edgecast/resources/cps/certificate.go:72]: Can remove this if it is to stay on line 50

**Review by wdawg33 — CHANGES_REQUESTED** [review]

**Review by ssaluja4 — APPROVED** [review]

File: edgecast/resources/cps/certificate_test.go [hunk 4]
```diff
@@ -435,3 +438,181 @@ func TestFlattenActor(t *testing.T) {
 		}
 	}
 }
+
+func TestExpandNotifSettings(t *testing.T) {
```
Comment (wdawg33) [inline edgecast/resources/cps/certificate_test.go:442]: Should we cover the cases that return an error here?

File: edgecast/resources/cps/certificate.go [hunk 5]
```diff
@@ -279,6 +349,126 @@ func ExpandDomains(attr interface{}) ([]*models.DomainCreateUpdate, error) {
...
+			errs = append(errs, err)
+		}
+
+		emailNotifs = append(emailNotifs, emailNotif)
+	}
+
+	return emailNotifs, errs
+}
+
+func GetStringFromMap(m map[string]any, key string) (string, bool) {
```
Comment (wdawg33) [inline edgecast/resources/cps/certificate.go:427]: Are you planning to add unit tests for these functions in the future (or are they moving out of certificate.go into helper?)

**Review by wdawg33 — CHANGES_REQUESTED** [review]

**Review by wdawg33 — APPROVED** [review]

**Pull request merged**